link.go

```diff
@@ -200,46 +229,62 @@ func linkPackage(hash string) (string, error) {
 		return "", fmt.Errorf("error during gx install: %s", err)
 	}
 
-	rwcmd := exec.Command("gx-go", "hook", "post-install", gxdir)
+	rwcmdArgs := []string{"hook", "post-install", linkPackageDir}
+	if sync {
+		rwcmdArgs = append(rwcmdArgs, "--sync", parentPackagePath)
+	}
+	rwcmd := exec.Command("gx-go", rwcmdArgs...)
 	rwcmd.Dir = target
 	rwcmd.Stdout = os.Stdout
 	rwcmd.Stderr = os.Stderr
 	if err := rwcmd.Run(); err != nil {
 		return "", fmt.Errorf("error during gx-go rw: %s", err)
 	}
+	// TODO: Wrap command calls in a function.
 
 	return target, nil
 }
 
+// Return the DVCS import path of a dependency (fetching it
+// if necessary).
+func findDepDVCSimport(dep *gx.Dependency, gxSrcDir string) (string, error) {
+	gxdir := filepath.Join(gxSrcDir, "gx", "ipfs", dep.Hash)
+
+	// Get the dependency to find out its DVCS import.
+	gxget := exec.Command("gx", "get", dep.Hash, "-o", gxdir)
+	gxget.Stdout = os.Stderr
+	gxget.Stderr = os.Stderr
+	if err := gxget.Run(); err != nil {
+		return "", fmt.Errorf("error during gx get: %s", err)
+	}
+
+	var pkg gx.Package
+	err := gx.FindPackageInDir(&pkg, gxdir)
+	if err != nil {
+		return "", fmt.Errorf("error during gx.FindPackageInDir: %s", err)
+	}
+
+	return GxDvcsImport(&pkg), nil
+}
+
 // rm -rf $GOPATH/src/gx/ipfs/$hash
 // gx get $hash
-func unlinkPackage(hash string) (string, error) {
-	srcdir, err := gx.InstallPath("go", "", true)
+func unlinkDependency(dep *gx.Dependency, parentPackagePath string) (string, error) {
+	gxSrcDir, err := gx.InstallPath("go", "", true)
 	if err != nil {
 		return "", err
 	}
-	gxdir := filepath.Join(srcdir, "gx", "ipfs", hash)
-
-	err = os.RemoveAll(gxdir)
+	dvcsImport, err := findDepDVCSimport(dep, gxSrcDir)
 	if err != nil {
-		return "", fmt.Errorf("error during os.RemoveAll: %s", err)
-	}
-
-	gxget := exec.Command("gx", "get", hash, "-o", gxdir)
-	gxget.Stdout = nil
-	gxget.Stderr = os.Stderr
-	if err = gxget.Run(); err != nil {
-		return "", fmt.Errorf("error during gx get: %s", err)
+		return "", fmt.Errorf("error trying to get the DVCS import of the dependeny %s: %s", dep.Name, err)
 	}
 
-	var pkg gx.Package
-	err = gx.FindPackageInDir(&pkg, gxdir)
+	err = os.RemoveAll(filepath.Join(parentPackagePath, "vendor", "gx", "ipfs", dep.Hash))
 	if err != nil {
-		return "", fmt.Errorf("error during gx.FindPackageInDir: %s", err)
+		return "", fmt.Errorf("error during os.RemoveAll: %s", err)
 	}
 
-	dvcsimport := GxDvcsImport(&pkg)
-	target := filepath.Join(srcdir, dvcsimport)
+	target := filepath.Join(gxSrcDir, dvcsImport)
 
 	uwcmd := exec.Command("gx-go", "uw")
```

Comment:
If `--sync` was used during the link process we may need to un-write with `--fix`.

The `--fix` could be added in the unlink for all cases as a general precautionary measure (at the cost of more processing time when it may not be needed, but to simplify the code as much as possible and do not add any sync detection logic).